lib/mux.js

```diff
@@ -79,7 +77,7 @@ Mux.prototype._readIncoming = function() {
   }
   else { // ran out of room
     assert(this.newStreams.length > 0, "Expect some new streams to remain");
-    this.oldStreams = this.oldStreams.concat(this.newStreams);
+    Array.prototype.push.apply(this.oldStreams, this.newStreams);
```

Comment:
Is there a reason why prototype push is used instead of `this.oldstreams.push`?

Reply:
Well I can not test it, but you could do a this.oldstreams.push.apply(this.newstreams)?

I think I did this to make it clear that it does not instantiate a intermediary Array

Reply:
https://stackoverflow.com/questions/15444242/why-doesnt-array-push-apply-work

I think this was the reason. I don't want to reference this.oldstreams twice.